source/class/qx/tool/compiler/app/WebFont.js

```diff
@@ -88,236 +88,166 @@ qx.Class.define("qx.tool.compiler.app.WebFont", {
      * Helper which triggers a local font analyze run.
      *
      * @param filename {String} Filename for the local font
-     * @return {Promise}
+     * @return {Map<String,String>} mapping of glyphs to codepoints
```

Comment:
no, the function is an `async` function so by definition it returns a promise behind the scenes - to change it as you describe would be to say that the function that returns a promise that resolves to a promise which resolves to a map.

Reply:
I disagree, pretty strongly, actually. There is no indication in the API Viewer that a method is `async`. Therefore, to know how to use the method -- to know that you must `await` its return value -- we must either indicate that it returns a Promise (which it does by virtue of being labeled `async`), or fix the API Viewer to indicate methods that are `async`. The latter isn't likely to happen any time soon. In the mean time, we should indicate in the JSDoc that methods that are `async` return a Promise.

Reply:
It's not that simple. There appears to be nothing in the .json file created by the compiler that indicates that the method is `async`, so the API viewer has no way to even be able to display it.

`async` is effectively syntactic sugar that says, "This method returns a Promise". Granted, it also enables the `await` keyword, so it's not quite syntactic sugar, but that's a big part of its purpose.

IMO, we are breaking usability of our documentation if we don't indicate that a Promise is returned by an async function *somehow*, and the only available way right now to do that is to say so in the JSDoc.

Reply:
> > It's not that simple. There appears to be nothing in the .json file created by the compiler
> 
> Why is that not fixable? The AST has AsyncFunction and Function. There is no reason why this cannot be added to the compiler output

Once it's added, I'd have no objection. Until it's added, I have an objection.

> 
> > async is effectively syntactic sugar that says, "This method returns a Promise".
> 
> Correct. But the documentation you are proposing says that "This method (which we know will implicitly return a Promise) will return a Promise which returns something else".

Close It says, "This method will return a Promise which *resolves to* something else", as shown in checked/accepted answer here: https://stackoverflow.com/questions/13104411/how-to-specify-resolution-and-rejection-type-of-the-promise-in-jsdoc

> 
> `async` functions are written on the basis of the Promise being implicit and you're proposing making it explicit in the documentation.

It can't be implicit. The user of the method needs to know that it's async to know how to use it. If they call it, now knowing it's async and thinking it actually returns a Map (rather than a Promise that resolves to a Map), they'll have a debugging task to ascertain that the documentation was inaccurate.

> 
> > IMO, we are breaking usability of our documentation if we don't indicate that a Promise is returned by an async function somehow, and the only available way right now to do that is to say so in the JSDoc.
> 
> Then the documentation is already broken because what you are proposing is 100% opposite to what we have done in the past. As I said, I've been asked to undo exactly what you have proposed in the past.

I don't recall any such discussions in the past. I would think I would have had similar comments at that time. Can you show me one of those discussions?

> 
> Fixing the API viewer will make your change something that needs to be reverted.

Agree. It's also not a big deal. There aren't that many uses of async methods. The ones that I looked at in the framework already properly document that they return a Promise. (They don't use the JSDoc indication of what the Promise resolves to; instead, they document the resolution in the description of the return value. That could be fixed if we want, too.) I'd much prefer to have accurate documentation now, while awaiting changes to the compiler and the API viewer at some point in the future, and have to fix the documentation once those things are updated, than to have inaccurate documentation until some unknown time when those will be updated.

Reply:
> I don't recall any such discussions in the past. I would think I would have had similar comments at that time. Can you show me one of those discussions?

I'm sure I have been asked to at some point, but I'm really not going back through every PR for years to try and find it.

> There aren't that many uses of async methods. The ones that I looked at in the framework already properly document that they return a Promise. 

Then you're not looking very far.  Searching for`async [a-zA-Z].*\(.*\)` locates 197 results in 89 files, the majority depend on `async` being a suitable marker, but there are a lot that also declare `Promise` in jsdoc.

Reply:
> async is effectively syntactic sugar that says, "This method returns a Promise".
> Correct. But the documentation you are proposing says that "This method (which we know will implicitly return a Promise) will return a Promise which returns something else".

>Close It says, "This method will return a Promise which _resolves_ to something else", as shown in

I'm not sure whether you are trying to pick me up on terminology here (FWIW I know that the correct word is "resolve" rather than "return"), or if you're trying to say that a StackOverflow answer supports your argument about how we should include `Promise` in the JSDoc, but I can't see that acceptance or relevance.

Why on earth are you creating yet another argument about some minutia?  I'm trying to solve a problem here, I am being consistent with the framework in general, and you are picking on one single line and demanding that it is changed for no substantial reason that I can see

Reply:
> I volunteer to review all 197 results at such time as the compiler and API viewer are updated to indicate that a method is `async`. Until then, my objection remains.

Assuming that you first fix the API viewer and review to remove the `Promise` in the JSDoc, then that'll be fine.